src/views/Profile.tsx

```
@@ -59,60 +44,67 @@ function Profile() {
     } else if (!isLoading) {
       navigate("/");
     }
-  }, [isLoading]);
-
-  const profilePictures: ProfilePicture[] = [
-    { name: "Blue Profile", path: "images/Blue_Profile.png" },
-    { name: "Red Profile", path: "images/Red_Profile.png" },
-    { name: "Green Profile", path: "images/Green_Profile.png" },
-    { name: "Pink Profile", path: "images/Pink_Profile.png" },
-    { name: "Yellow Profile", path: "images/Yellow_Profile.png" },
-    { name: "Purple Profile", path: "images/Purple_Profile.png" },
-  ];
-
-  const handleProfilePictureClick = () => {
-    setProfilePictureClicked(!profilePictureClicked);
-  };
+  }, [isLoading, isAuthenticated, getToken, navigate]);
+
+  const handleProfilePictureUpload = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file && savedToken) {
+      if (!file.type.startsWith("image/")) {
+        alert("Please upload a valid image file.");
+        return;
+      }
+
+      const formData = new FormData();
+      formData.append("profilePicture", file);
+
+      try {
+        const response = await fancyFetch({
+          endpoint: "/users/profilepicture",
+          method: "POST",
+          data: formData,
+          token: savedToken,
+        });
+
+        if (!response.ok) {
+          throw new Error("Failed to upload profile picture");
```

I am not sure if in our current architecture you want to throw an error here. I think alerting the user is a better approach since errors will just be put on the console.